CV Input

arjunsudheer opened the pull request:
Get ball distance and angle information

arjunsudheer committed:
Add code to get the ball information

Get ball distance and angle information

arjunsudheer committed:
Add code to get the ball information

Get ball distance and angle information

Remove class structure. Update data_sender.py

Add timer and update color and ball detection calls.

Fix comment formatting

Add serial port and data sending code

Open serial port and send ball distance and angle as a bytearray

Remove unneeded multi-processing code

Make CV code run indefinitely.

Remove timer restriction.

Pass data to interface instead of serial port

Pass ball data and serial object to interface

Update import statement

Rename encoding import statement for compatability with interface

Remove unneeded imports

arjunsudheer committed:
Remove class structure. Update data_sender.py

Add timer and update color and ball detection calls.

arjunsudheer committed:
Fix comment formatting

arjunsudheer committed:
Add serial port and data sending code

Open serial port and send ball distance and angle as a bytearray

arjunsudheer committed:
Remove unneeded multi-processing code

AG6GR commented on data_sender.py:
```diff
@@ -1,11 +1,34 @@
+print(ser.name)
+
+
+# run the cv code for 150 seconds (15 for autonomous + 125 for TeleOp)
```
We should run the CV code indefinitely. The robot gets powered on when we put it on the field, and the Jetson will start running CV then. We don't know how long it will be until the match actually starts.

arjunsudheer replied:
Thank you for the feedback. I have updated it.

arjunsudheer committed:
Make CV code run indefinitely.

Remove timer restriction.

arjunsudheer committed:
Pass data to interface instead of serial port

Pass ball data and serial object to interface

arjunsudheer committed:
Update import statement

Rename encoding import statement for compatability with interface

DattaKansal committed:
Interface (#9)

* Added encoding file. Not tested yet.

* updating documents

* Add files via upload

* Update Encoding.py

* Updated python style checks

* Modified send_data code

* Fixed python style check

* renamed file

* modified according to Decoding standards.

* remove init function and .idea file

arjunsudheer committed:
Remove unneeded imports

ukshat committed:
Cvdetection (#5)

* Create color_detector.py

* Update color_detector.py

* Added test images, found thresholds

* Create CVDetectionSpec.md

* Created 12 binary images from test images

* Fixed style issues

* Fixed style issues

* Simplified masking code

* Moved images from GitHub to Google Drive

* Reformatted code into multiple files

* Created quantitative accuracy measurement

* Updated detect() parameters

* Removed excess measure_accuracy images

* Added detect_and_save() method, ran it on the 55 images

* Added blur, added second mask for red

* Added blank image for bitwise_and

* Added functionality to iterate through images

* Implemented new blurring algorithms (will be used for shadow removal)

* Shadow removal code using YUV

* Created color dector using YUV

* Finalized YUV bounds

* Measured for 22 images on HSV

* switched to YUV Color Detector

* Added code for total accuracy

* Create color_detector.py

* Update color_detector.py

* Added test images, found thresholds

* Created 12 binary images from test images

* Create CVDetectionSpec.md

* Fixed style issues

* Fixed style issues

* Simplified masking code

* Moved images from GitHub to Google Drive

* Reformatted code into multiple files

* Created quantitative accuracy measurement

* Updated detect() parameters

* Removed excess measure_accuracy images

* Added detect_and_save() method, ran it on the 55 images

* Added blur, added second mask for red

* Added blank image for bitwise_and

* Added functionality to iterate through images

* Implemented new blurring algorithms (will be used for shadow removal)

* Shadow removal code using YUV

* Created color dector using YUV

* Finalized YUV bounds

* Measured for 22 images on HSV

* switched to YUV Color Detector

* Calculated YUV bounds

* Finalized blue YUV bounds

* Removed .jpg files

* Fixed Python style issues

* Trying out HoughCircles

* Renamed files

* Removed excess files

* Added HoughCircles code

* Removed HoughCircles

* Replaced os.listdir with pathlib

* Renamed ColorDetectorHSV

* Removed HoughCircles code

* Replaced blank.jpg with np.full()

* added argparse

* HSV instead of YUV for blue

* Removed shadow_removal.py

Co-authored-by: Ishani Das <35828031+ishani-das@users.noreply.github.com>
Co-authored-by: Ishani Das <ishanidas05@gmail.com>

arjunsudheer committed:
LimeLight Code (#8)

* Add initial code to run on LimeLight.

Bring existing color and ball detection code. Add image annotation code.

* Remove camera setup code. Update output tuple.

* Update color and ball detection code

Update to YUV, remove imutils import

* Fix pycodestyle errors

* Fixed depth issue

* Update image annotations.

Fix positioning, and font weight.

* Interface (#9)

* Added encoding file. Not tested yet.

* updating documents

* Add files via upload

* Update Encoding.py

* Updated python style checks

* Modified send_data code

* Fixed python style check

* renamed file

* modified according to Decoding standards.

* remove init function and .idea file

* Add NetworkTables for alliance color specification

* Fix pycodestyle errors

* Update ball detection code, add llrobot code

Add code to get alliance color from drive team

* Fix pycodestyle errors

* Add initial code to run on LimeLight.

Bring existing color and ball detection code. Add image annotation code.

Remove camera setup code. Update output tuple.

Update color and ball detection code

Update to YUV, remove imutils import

Fix pycodestyle errors

Fixed depth issue

Update image annotations.

Fix positioning, and font weight.

Add NetworkTables for alliance color specification

Fix pycodestyle errors

Update ball detection code, add llrobot code

Add code to get alliance color from drive team

Fix pycodestyle errors

* Cvdetection (#5)

* Create color_detector.py

* Update color_detector.py

* Added test images, found thresholds

* Create CVDetectionSpec.md

* Created 12 binary images from test images

* Fixed style issues

* Fixed style issues

* Simplified masking code

* Moved images from GitHub to Google Drive

* Reformatted code into multiple files

* Created quantitative accuracy measurement

* Updated detect() parameters

* Removed excess measure_accuracy images

* Added detect_and_save() method, ran it on the 55 images

* Added blur, added second mask for red

* Added blank image for bitwise_and

* Added functionality to iterate through images

* Implemented new blurring algorithms (will be used for shadow removal)

* Shadow removal code using YUV

* Created color dector using YUV

* Finalized YUV bounds

* Measured for 22 images on HSV

* switched to YUV Color Detector

* Added code for total accuracy

* Create color_detector.py

* Update color_detector.py

* Added test images, found thresholds

* Created 12 binary images from test images

* Create CVDetectionSpec.md

* Fixed style issues

* Fixed style issues

* Simplified masking code

* Moved images from GitHub to Google Drive

* Reformatted code into multiple files

* Created quantitative accuracy measurement

* Updated detect() parameters

* Removed excess measure_accuracy images

* Added detect_and_save() method, ran it on the 55 images

* Added blur, added second mask for red

* Added blank image for bitwise_and

* Added functionality to iterate through images

* Implemented new blurring algorithms (will be used for shadow removal)

* Shadow removal code using YUV

* Created color dector using YUV

* Finalized YUV bounds

* Measured for 22 images on HSV

* switched to YUV Color Detector

* Calculated YUV bounds

* Finalized blue YUV bounds

* Removed .jpg files

* Fixed Python style issues

* Trying out HoughCircles

* Renamed files

* Removed excess files

* Added HoughCircles code

* Removed HoughCircles

* Replaced os.listdir with pathlib

* Renamed ColorDetectorHSV

* Removed HoughCircles code

* Replaced blank.jpg with np.full()

* added argparse

* HSV instead of YUV for blue

* Removed shadow_removal.py

Co-authored-by: Ishani Das <35828031+ishani-das@users.noreply.github.com>
Co-authored-by: Ishani Das <ishanidas05@gmail.com>

* Add initial code to run on LimeLight.

Bring existing color and ball detection code. Add image annotation code.

Remove camera setup code. Update output tuple.

Update color and ball detection code

Update to YUV, remove imutils import

Fix pycodestyle errors

Fixed depth issue

Update image annotations.

Fix positioning, and font weight.

Add NetworkTables for alliance color specification

Fix pycodestyle errors

Update ball detection code, add llrobot code

Add code to get alliance color from drive team

Fix pycodestyle errors

Update ball detection code, add llrobot code

Add code to get alliance color from drive team

Fix pycodestyle errors

Co-authored-by: ukshat <akshat@punatar.org>
Co-authored-by: Datta Kansal <55328687+LegendaryDK@users.noreply.github.com>
Co-authored-by: Akshat Mehta <33008157+ukshat@users.noreply.github.com>
Co-authored-by: Ishani Das <35828031+ishani-das@users.noreply.github.com>
Co-authored-by: Ishani Das <ishanidas05@gmail.com>

arjunsudheer committed:
Merge branch 'cv-input_multi' of github.com:Tino-FRC-2473/FRC2022-Vision into cv-input_multi
